src/messages/create/publish.ts

```diff
@@ -75,20 +76,28 @@ async function PushToStorageEngine<T>(configuration: PushConfiguration<T>): Prom
             },
         },
     );
-
     return response.data.hash;
 }
 
 export async function PushFileToStorageEngine(configuration: PushFileConfiguration): Promise<string> {
-    const form = new FormData();
+    const isBrowser = typeof FormData !== "undefined";
+    let form: FormDataNode | FormData;
 
-    form.append("file", configuration.file);
+    if (isBrowser) {
+        form = new FormData();
+        form.append("file", new Blob([configuration.file]));
+    } else {
+        form = new FormDataNode();
+        form.append("file", configuration.file, "usageName.txt");
```

Comment:
Is it a normal behaviour to have this hardcoded string ?

Reply:
This last parameter represents the fileName according to the [documentation](https://developer.mozilla.org/fr/docs/Web/API/FormData/append).

It does not appear to be mandatory, but I got an error 500 when I don't mention it.
So I changed `"usageName.txt"` by the default value `"File"`=> 709c021